node/src/tasks/msp_charge_fees.rs

```diff
@@ -0,0 +1,129 @@
+        }
+    }
+}
+
+impl<FL, FSH> MspChargeFeesTask<FL, FSH>
+where
+    FL: FileStorageT,
+    FSH: MspForestStorageHandlerT,
+{
+    pub fn new(storage_hub_handler: StorageHubHandler<FL, FSH>) -> Self {
+        Self {
+            storage_hub_handler,
+        }
+    }
+}
+
+/// Handles the [`NotifyPeriod`] event.
+///
+/// This event is triggered every X amount of blocks.
+///
+/// This task will:
+/// - Charge users for the MSP when triggered
+impl<FL, FSH> EventHandler<NotifyPeriod> for MspChargeFeesTask<FL, FSH>
+where
+    FL: FileStorageT,
+    FSH: MspForestStorageHandlerT,
+{
+    async fn handle_event(&mut self, _event: NotifyPeriod) -> anyhow::Result<()> {
+        info!(
+            target: LOG_TARGET,
+            "Charging users",
+        );
+
+        let own_provider_id = self
+            .storage_hub_handler
+            .blockchain
+            .query_storage_provider_id(None)
+            .await?;
+
+        let own_msp_id = match own_provider_id {
+            Some(id) => match id {
+                StorageProviderId::MainStorageProvider(id) => id,
+                StorageProviderId::BackupStorageProvider(_) => {
+                    let err_msg = "Current node account is a Backup Storage Provider. Expected a Main Storage Provider ID.";
+                    error!(target: LOG_TARGET, err_msg);
+                    return Err(anyhow!(err_msg));
+                }
+            },
+            None => {
+                let err_msg = "Failed to get own MSP ID.";
+                error!(target: LOG_TARGET, err_msg);
+                return Err(anyhow!(err_msg));
+            }
+        };
+
+        let users_with_debt = self
+            .storage_hub_handler
+            .blockchain
+            .query_users_with_debt(own_msp_id, MIN_DEBT)
+            .await
+            .map_err(|e| {
+                anyhow!(
+                    "Failed to retrieve users with debt from the runtime: {:?}",
+                    e
+                )
+            })?;
+
+        // Divides the users to charge in chunks of MaxUsersToCharge to avoid exceeding the block limit.
+        // Calls the `charge_multiple_users_payment_streams` extrinsic for each chunk in the list to be charged.
+        // Logs an error in case of failure and continues.
+        let user_chunk_size: u32 = MaxUsersToCharge::get();
+        for users_chunk in users_with_debt.chunks(user_chunk_size as usize) {
+            let call = storage_hub_runtime::RuntimeCall::PaymentStreams(
+                pallet_payment_streams::Call::charge_multiple_users_payment_streams {
+                    user_accounts: users_chunk.to_vec().try_into().expect("Chunk size is the same as MaxUsersToCharge, it has to fit in the BoundedVec"),
+                },
+            );
+
+            let charging_result = self
+                .storage_hub_handler
+                .blockchain
+                .send_extrinsic(call, Tip::from(0))
```

Comment:
We may want to use `watch_for_success` here.

Reply:
For BSP, we are not watching for success. Do we want to have it there too ?